File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-3.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n = arr.size();
        int max = -1;
        for(int i = n-1; i>=0;i--){
            int tmp = arr[i];
            arr[i] = max;
            if(tmp > max){
                max = tmp;
            }
            
        }
        return arr;
    }
};

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:

        cur = root
        while cur:
            if cur.val > max(p.val,q.val):
                cur = cur.left
            elif cur.val < min(p.val,q.val):
                cur = cur.right
            else:
                return cur